lib/s3routes/routesUtils.js

```diff
@@ -85,8 +85,17 @@ const XMLResponseBackend = {
         });
     },
 
-    errorResponse: function errorXMLResponse(errCode, response, log,
-                                             corsHeaders) {
+    errorResponse: function errorXMLResponse(errCode, response, log, corsHeaders) {
+        setCommonResponseHeaders(corsHeaders, response, log);
+        if (errCode.code === 304) {
```

Comment:
may be add a comment why we are doing this!

Reply:
I'd recommend adding the JIRA ticket number of the bugfix too in your comment like 
// [S3C-3810] ....reason...